src/main/java/org/commcare/formplayer/services/CategoryTimingHelper.java

```diff
@@ -66,31 +75,38 @@ public void recordCategoryTiming(Timing timing, String category) {
     public void recordCategoryTiming(Timing timing, String category, String sentryMessage) {
         recordCategoryTiming(timing, category, sentryMessage, null);
     }
-    public void recordCategoryTiming(Timing timing, String category, String sentryMessage, String domain) {
+
+    /**
+     * @param extras - optional arguments, the following keys are accepted:
+     *               CategoryTimingHelper.DOMAIN,
+     *               CategoryTimingHelper.FORM_NAME,
+     */
+    public void recordCategoryTiming(Timing timing, String category, String sentryMessage, Map<String, String> extras) {
         raven.newBreadcrumb()
                 .setCategory(category)
                 .setMessage(sentryMessage)
                 .setData("duration", timing.formatDuration())
                 .record();
 
-        if (domain != null) {
-            datadogStatsDClient.recordExecutionTime(
-                Constants.DATADOG_GRANULAR_TIMINGS,
-                timing.durationInMs(),
-                "category:" + category,
-                "request:" + RequestUtils.getRequestEndpoint(request),
-                "duration:" + timing.getDurationBucket(),
-                "domain:" + domain
-            );
-        } else {
-            datadogStatsDClient.recordExecutionTime(
+        List<String> datadogArgs = new ArrayList<>();
+        datadogArgs.add("category:" + category);
+        datadogArgs.add("request:" + RequestUtils.getRequestEndpoint(request));
+        datadogArgs.add("duration:" + timing.getDurationBucket());
+        if (extras != null) {
```

Comment:
Is it possible to just iterate over the map and add those tags rather than check for known keys? Then future updates won't have to update this method as well and we can add arbitrary tags at any point.

Reply:
Yeah great point.

Reply:
This will be somewhat moot once the DatadogWrapper is used since we will likely just add the tag prior to this method, but I think this functionality is still useful if wanting to easily add a tag for a specific call.